pkg/apis/autoscaling/v1alpha1/pa_lifecycle.go

```diff
@@ -148,21 +148,24 @@ func (pa *PodAutoscaler) PanicThresholdPercentage() (percentage float64, ok bool
 
 // IsReady looks at the conditions and if the Status has a condition
 // PodAutoscalerConditionReady returns true if ConditionStatus is True
-func (pas *PodAutoscalerStatus) IsReady() bool {
-	return podCondSet.Manage(pas).IsHappy()
+func (pa *PodAutoscaler) IsReady() bool {
+	pas := pa.Status
+	return pa.Generation == pas.ObservedGeneration && podCondSet.Manage(&pas).IsHappy()
 }
 
 // IsActivating returns true if the pod autoscaler is Activating if it is neither
 // Active nor Inactive
-func (pas *PodAutoscalerStatus) IsActivating() bool {
-	cond := pas.GetCondition(PodAutoscalerConditionActive)
-	return cond != nil && cond.Status == corev1.ConditionUnknown
+func (pa *PodAutoscaler) IsActivating() bool {
+	pas := pa.Status
+	return pa.Generation != pas.ObservedGeneration ||
```

Comment:
Should this also include the case of generations matching? This might affect the reason of PA inactiveness https://github.com/knative/serving/blob/master/pkg/reconciler/autoscaling/kpa/kpa.go#L256

Reply:
If generation matches observed, we will have set the status of the condition to "Unknown" during reconciliation if we are still activating.

Reply:
I think this is correct, but let's split out generation for conditions into another PR